### DISCUSSION TITLE
How to contribute a Custom Quirk?

### DISCUSSION THREAD

**Discussion by AlexW75** [zigpy/zha-device-handlers#3215] [Q&A]
Hi, 
I've created my first custom quirk for the Phoscon HIVE, a Zigbee controllable battery lamp. 
The issue with ZHA is that it doesn't display the battery status. My quirk resolves this problem. 

How can I provide it?

**Comment by Hedda** [comment]
https://github.com/zigpy/zha-device-handlers/blob/dev/README.md

Suggest you always begin by creating an new issue regardless as helps with tracking and any device-specific discussion before create PR:

https://github.com/zigpy/zha-device-handlers/issues

More generic tips and links on how to make a contribution on GitHub as a developer is available n this similar project here:

https://github.com/zigpy/zigpy/blob/dev/CONTRIBUTING.md

Especially sstart by reading and following links in this section:

## How to contribute as a developer

If you are looking to make a contribution as a developer to this project we suggest that you follow the steps in these guides:

- https://github.com/firstcontributions/first-contributions/blob/master/README.md
  - https://github.com/firstcontributions/first-contributions/blob/master/github-desktop-tutorial.md

Code changes or additions can then be submitted to this project on GitHub via pull requests:

- https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-requests
  - https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request

In general when contributing code to this project it is encouraged that you try to follow the coding standards:

- First [raise issues on GitHub](https://github.com/zigpy/zigpy/issues) before working on an enhancement to provide coordination with other contributors.
- Try to keep each pull request short and only a single PR per enhancement as this makes tracking and reviewing easier.
- All code is formatted with black. The check format script that runs in CI will ensure that code meets this requirement and that it is correctly formatted with black. Instructions for installing black in many editors can be found here: https://github.com/psf/black#editor-integration
- Ideally, you should aim to achieve full coverage of any code changes with tests.
- Recommend read and follow [Google Python Style Guide](https://google.github.io/styleguide/pyguide.html).
- Recommend read and follow [Clifford Programming Style](http://www.clifford.at/style.html).
- Recommend code style use [standard naming conventions for Python](https://medium.com/@dasagrivamanu/python-naming-conventions-the-10-points-you-should-know-149a9aa9f8c7).
- Recommend use [Semantic Versioning](http://semver.org/) for libraries and dependencies if possible.
- Contributions must be your own and you must agree with the license.
- All code for this project should aim to be licensed under [GNU GENERAL PUBLIC LICENSE Version 3](https://raw.githubusercontent.com/zigpy/zigpy/dev/LICENSE).
